src/Microsoft.AspNetCore.Blazor.Razor.Extensions/BlazorDesignTimeNodeWriter.cs

```diff
@@ -0,0 +1,499 @@
+                    // There may be something else inside the statement like an extension node.
+                    context.RenderNode(node.Children[i]);
+                }
+            }
+        }
+
+        public override void WriteHtmlContent(CodeRenderingContext context, HtmlContentIntermediateNode node)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            if (node == null)
+            {
+                throw new ArgumentNullException(nameof(node));
+            }
+
+            // Do nothing
+        }
+
+        public override void BeginWriteAttribute(CodeWriter codeWriter, string key)
+        {
+            codeWriter
+                .WriteStartMethodInvocation($"{_scopeStack.BuilderVarName}.{nameof(BlazorApi.RenderTreeBuilder.AddAttribute)}")
+                .Write("-1")
+                .WriteParameterSeparator()
+                .WriteStringLiteral(key)
+                .WriteParameterSeparator();
+        }
+
+        public override void WriteComponentOpen(CodeRenderingContext context, ComponentOpenExtensionNode node)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            if (node == null)
+            {
+                throw new ArgumentNullException(nameof(node));
+            }
+
+            // Do nothing
+        }
+
+        public override void WriteComponentClose(CodeRenderingContext context, ComponentCloseExtensionNode node)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            if (node == null)
+            {
+                throw new ArgumentNullException(nameof(node));
+            }
+
+            // Do nothing
+        }
+
+        public override void WriteComponentBody(CodeRenderingContext context, ComponentBodyExtensionNode node)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            if (node == null)
+            {
+                throw new ArgumentNullException(nameof(node));
+            }
+
+            // We need to be aware of the blazor scope-tracking concept in design-time code generation
+            // because each component creates a lambda scope for its child content.
+            //
+            // We're hacking it a bit here by just forcing every component to have an empty lambda
+            _scopeStack.OpenScope(node.TagName, isComponent: true);
+            _scopeStack.IncrementCurrentScopeChildCount(context);
+            context.RenderChildren(node);
+            _scopeStack.CloseScope(context, node.TagName, isComponent: true, source: node.Source);
+        }
+
+        public override void WriteComponentAttribute(CodeRenderingContext context, ComponentAttributeExtensionNode node)
```

Comment:
@NTaylorMullen @ajaybhargavb - I feel like I was able to really simplify this code from what's currently the state of the art in Razor. I also used this opportunity to prototype and approach for Razor's design time code that was suggested by @jasonmalinowski to solve our alignment woes in the debugger...

Example:
```
            __o = 
#line 6 "x:\dir\subdir\Test\TestComponent.cshtml"
                    new SomeType()

#line default
#line hidden
            ;
```
Traditionally we'd put all of this junk on one line. This is a problem whenever the design time code gen has a significant offset that differs from what the user typed.

This deffo comes up in Blazor with cases like the following - imagine the user's code is in the `...`.
```
#line 6 "x:\dir\subdir\Test\TestComponent.cshtml"
            __o =  new UIEventHandler(e => ...);
#line default
#line hidden
```

Reply:
When we initially re-did Razor for .net Core this is what I did. In the end it ended up breaking formatting and a few other pieces that I can't recall. Because of that we ended up reverting the multi-line code gen functionality 😢